src/provider.rs

```diff
@@ -2,13 +2,30 @@
 
 use core::str::FromStr;
 
+use crate::UtcOffset;
 use crate::{iso::IsoDateTime, unix_time::EpochNanoseconds, TemporalResult};
 use alloc::borrow::Cow;
 
 /// `UtcOffsetSeconds` represents the amount of seconds we need to add to the UTC to reach the local time.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
 pub struct UtcOffsetSeconds(pub i64);
 
+impl From<UtcOffsetSeconds> for UtcOffset {
+    fn from(other: UtcOffsetSeconds) -> Self {
+        Self::from_seconds(other.0)
+    }
+}
+
+/// An EpochNanoseconds and a UTC offset
+#[derive(Copy, Clone, Debug)]
+pub struct EpochNanosecondsAndOffset {
```

Comment:
praise: I actually like this new type!

I think the extra information that it provides is probably a good thing. (not exactly in love with the name though).